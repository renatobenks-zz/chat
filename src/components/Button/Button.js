// @flow

import * as React from 'react';
import styled from 'styled-components';

import type { Interpolation, TaggedTemplateLiteral } from 'styled-components';

const CustomButton = styled.button`
  margin: 1rem;
  padding: ${props => props.theme.padding};
  background: ${props => props.background || 'transparent'};
  color: ${props => props.color || props.theme.palette.primary};
  border: none;
  cursor: pointer;

  svg {
    fill: ${props => props.color || props.theme.palette.primary};
  }

  ${props => props.styles};
`;

type Props = {
  children: React.Element<any>,
  styles?: TaggedTemplateLiteral<Array<Interpolation>>,
  background?: ?string,
  color?: ?string,
  onClick: () => void,
};

const Button = ({
  children, styles, background, color, onClick 
}: Props) => {
  return (
    <CustomButton background={background} color={color} styles={styles} onClick={onClick}>
      {children}
    </CustomButton>
  );
};

export default Button;
