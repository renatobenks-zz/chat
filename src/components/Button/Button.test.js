import React from 'react';
import { css } from 'styled-components';

import { render } from '../../../test/helpers';

import Button from './Button';

const renderer = component => render(component, { withTheme: true });

it('button', () => {
  expect(
    renderer(
      <Button>
        <span>message</span>
      </Button>,
    ),
  ).toMatchSnapshot();
});

it('button with custom background', () => {
  expect(
    renderer(
      <Button background="black">
        <span>message</span>
      </Button>,
    ),
  ).toMatchSnapshot();
});

it('button with custom color', () => {
  expect(renderer(<Button color="white" />)).toMatchSnapshot();
});

it('button with custom styles', () => {
  expect(
    renderer(
      <Button
        styles={css`
          margin: 1rem 0;
        `}
      />,
    ),
  ).toMatchSnapshot();
});

it('handle button click', () => {
  const handler = jest.fn();
  const tree = renderer(<Button onClick={handler} />).toJSON();
  expect(tree).toMatchSnapshot();
  tree.props.onClick();
  expect(handler).toBeCalled();
});
