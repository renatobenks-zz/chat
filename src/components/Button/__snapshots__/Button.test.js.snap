// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 1`] = `
<button
  className="sc-bdVaJa dPPcZB"
>
  <span>
    message
  </span>
</button>
`;

exports[`button with custom background 1`] = `
<button
  className="sc-bdVaJa fEIDvw"
>
  <span>
    message
  </span>
</button>
`;

exports[`button with custom color 1`] = `
<button
  className="sc-bdVaJa kvxBxa"
  color="white"
/>
`;

exports[`button with custom styles 1`] = `
<button
  className="sc-bdVaJa kSjDXp"
/>
`;

exports[`handle button click 1`] = `
<button
  className="sc-bdVaJa dPPcZB"
  onClick={[MockFunction]}
/>
`;
